validator/client/validator_propose.go

```diff
@@ -59,6 +73,12 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 	}
 	b.Signature = sig
 
+	db.SetProposedForEpoch(history, epoch)
+	if err := v.db.SaveProposalHistory(pubKey[:], history); err != nil {
+		log.WithError(err).Error("Failed to save updated proposal history")
```

Comment:
This is very scary if it happens, is returning good enough here? should it keep reattempting to save a few times? what do you think?

Reply:
Good point, what kind of pattern can we do here?

Reply:
This will kill propose block as a whole, it needs a better solution